tests/test_gpu_common.py

```diff
@@ -1128,6 +1128,60 @@ def test_xcor_from_saved(self, opt, opt_options, gpu_fit):
 
         assert np.all(g.data == 30)
 
+    def test_gpu_create_forward(self):
+        nt = 10
+        grid = Grid(shape=(4, 4))
+
+        u = TimeFunction(name='u', grid=grid)
+        usave = TimeFunction(name='usave', grid=grid, save=nt)
+
+        for i in range(nt):
+            usave.data[i, :] = i
+
+        eqn = Eq(u.forward, u + usave)
+
+        op = Operator(eqn,
+                      opt=('buffering', 'streaming', 'orchestrate', {'gpu-create': u}))
+
+        language = configuration['language']
+        if language == 'openacc':
+            assert 'create(u' in str(op)
+        elif language == 'openmp':
+            assert 'map(alloc: u' in str(op)
+        assert 'init0(u_vec' in str(op)
+
+        op.apply(time_M=nt - 2)
+
+        assert np.all(u.data[0] == 28)
+        assert np.all(u.data[1] == 36)
+
+    def test_gpu_create_backward(self):
+        nt = 10
+        grid = Grid(shape=(4, 4))
+
+        u = TimeFunction(name='u', grid=grid)
+        usave = TimeFunction(name='usave', grid=grid, save=nt)
+
+        for i in range(nt):
+            usave.data[i, :] = i
+
+        eqn = Eq(u.backward, u + usave)
+
+        op = Operator(eqn,
+                      opt=('buffering', 'streaming', 'orchestrate', {'gpu-create': u}))
+
+        language = configuration['language']
+        if language == 'openacc':
+            assert 'create(u' in str(op)
+        elif language == 'openmp':
+            assert 'map(alloc: u' in str(op)
+        assert 'init0(u_vec' in str(op)
+
+        op.apply(time_M=nt - 2)
+
+        assert np.all(u.data[0] == 36)
+        assert np.all(u.data[1] == 35)
+
```

Comment:
Should `place-transfers` be tested out as well separatrly

Reply:
I can do that. Should I add the test to `test_gpu_common.py::TestStreaming`?